pkg/kudoctl/kudoinit/prereq/webhook.go

```diff
@@ -112,15 +115,19 @@ func (k KudoWebHook) Resources() []runtime.Object {
 	return k.resourcesWithCertManager()
 }
 
-func (k KudoWebHook) resourcesWithCertManager() []runtime.Object {
+func (k *KudoWebHook) resourcesWithCertManager() []runtime.Object {
+	// We have to fall back to a default here as for a dry-run we can't detect the actual version of a cluster
+	k.issuer = issuer(k.opts.Namespace, certManagerNewGroup, certManagerAPIVersions[0])
+	k.certificate = certificate(k.opts.Namespace, certManagerNewGroup, certManagerAPIVersions[0])
```

Comment:
these lines of code... can we get rid of them?   Issuer and Certificate do NOT seem to be installed with `--unsafe-self-signed-webhook-ca`. it is unclear if they are ever used... other than `--dry-run -o yaml`.   If we have a need to provide details around issuer and cert for this model of installation we should do it.   If these are needed for a specific style of installation we need to talk about it.

These should be complete removed which will resolve the remaining blocking issue.

Reply:
Well, they are not used when `--unsafe-self-signed-webhook-ca` is used. `Issuer` and `Certificate` are created when we don't use a self-signed-ca.

With this refactoring they are detected in the `preInstallVerify` step and the correct version of `Issuer` and `Certificate` for the installed cert-manager are created.

They are then applied in the `Install` step.

The problem with `resources` is that this step currently does not use the `preInstallVerify` step, and it usually doesn't even have a client initialized because it doesn't need an existing K8s cluster at all. Now that we need to detect the correct cert-manager version, we may have to do that. Maybe that's not a bad thing...